When removing a kernel it would be good to remove all the modules and not just uninstall them.
Issuing a remove instead of an uninstall to dkms will cause the module to be uninstalled first
(if it's not already), then removed. If the kernel is ever re-installed then dkims will
rebuild the module as needed, so nothing is lost by removing the modules for a
reomoved kernel, but you do run a cleaner system.

--- a/kernel_prerm.d_dkms
+++ b/kernel_prerm.d_dkms
@@ -8,8 +8,8 @@ while read line; do
    name=`echo "$line" | awk '{print $1}' | sed 's/,$//'`
    vers=`echo "$line" | awk '{print $2}' | sed 's/,$//'`
    arch=`echo "$line" | awk '{print $4}' | sed 's/:$//'`
-   echo "dkms: uninstalling: $name $vers ($inst_kern) ($arch)" >&2
-   dkms uninstall -m $name -v $vers -k $inst_kern -a $arch
+   echo "dkms: removing: $name $vers ($inst_kern) ($arch)" >&2
+   dkms remove -m $name -v $vers -k $inst_kern -a $arch
 done < <(dkms status -k $inst_kern 2>/dev/null | grep ": installed")
 fi
 
